sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorAdministrationClient.cs

```diff
@@ -291,6 +291,62 @@ public virtual Response<DataFeed> CreateDataFeed(string dataFeedName, DataFeedSo
             }
         }
 
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="dataFeedId"></param>
+        /// <param name="dataFeed"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response> UpdateDataFeedAsync(string dataFeedId, DataFeed dataFeed, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(dataFeedId, nameof(dataFeedId));
+            Argument.AssertNotNull(dataFeed, nameof(dataFeed));
+
+            Guid dataFeedGuid = ClientCommon.ValidateGuid(dataFeedId, nameof(dataFeedId));
+
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorAdministrationClient)}.{nameof(UpdateDataFeed)}");
+            scope.Start();
+            try
+            {
+                var patchModel = DataFeedDetail.GetPatchModel(dataFeed);
```

Comment:
if for whatever reason this fails, will the user get a useful message on why and how to fix it? I didn't see it in `GetPatchModel` and because of constructors I wonder if the error is just a generic one

Reply:
Generally the service returns a helpful error string in the exception.